crates/net/src/http/headers.rs

```diff
@@ -37,14 +44,14 @@ impl Headers {
 
     /// This method appends a new value onto an existing header, or adds the header if it does not
     /// already exist.
-    pub fn append(&self, name: &str, value: &str) {
+    pub fn append(&mut self, name: &str, value: &str) {
```

Comment:
I'm not a fan of making this `&mut`. This makes the method unnecessarily restrictive.

CC: @futursolo, do you have any thoughts on this?